### PR TITLE
[proposal] npm to pnpm

### PR DESCRIPTION
**Pull request by alewin**
## Motivation

https://pnpm.io/motivation

some other repo like:
- https://github.com/ogcio/govie-ds/
- https://github.com/ogcio/logto
- https://github.com/ogcio/analytics

are using `pnpm` without problems, leveraging its benefits

### CODE REVIEW

**Review by bozzelliandrea — APPROVED** [review]
100% agree

**Review by andreacappadona17 — APPROVED** [review]
LGTM

**Comment by andreacappadona17** [comment]
@SamSalvatico @relvao @alfonsograziano 
Any cons from your side?

**Comment by SamSalvatico** [comment]
> @SamSalvatico @relvao @alfonsograziano Any cons from your side?

I'm fine with this, the only question I have (because of my lack of knowledge) is: if we want to insert it into `life-events`, can we manage monorepo in the same way we did so far?

**Comment by alfonsograziano** [comment]
> @SamSalvatico @relvao @alfonsograziano Any cons from your side?

No blockers for me

**Comment by alewin** [comment]
> > @SamSalvatico @relvao @alfonsograziano Any cons from your side?
> 
> I'm fine with this, the only question I have (because of my lack of knowledge) is: if we want to insert it into `life-events`, can we manage monorepo in the same way we did so far?

what do you mean specifically?

`pnpm link` is supported https://pnpm.io/cli/link as well and on [analytics](https://github.com/ogcio/analytics/blob/main/pnpm-workspace.yaml) we are using pnpm as a "monorepo manager", which in my opinion is more convenient than npm workspace
if I've misunderstood, we can get in touch 👌

**Review by matteolc — APPROVED** [review]
👍🏽 LGTM
Maybe remove `run` from `pnpm` commands where not strictly required

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,2 +1,2 @@
-npm run check:formatting
-npm run check:linting
+pnpm run check:formatting
+pnpm run check:linting
```
[the remark this answers is not in the document]
Comment (alewin) [reply .husky/pre-commit:2]: thanks, if there aren't further comments, i'll make the changes before merging it

**Pull request merged**